### ISSUE TITLE
Help understanding "Missing Sender" when trying to send password reset email

### ISSUE THREAD

**Issue by itispeach** [firebase/firebaseui-web#80]
[ Warning : Designer trying to code]

When I try to send a password reset email, I see the "missing sender" alert in my window and the "auth/invalid-email" error in my console to go along with it. I suspect that this has something to do with my email configuration. I did my darnedest to customize my email domain. I followed the steps in the firebase console, and my email client was verified. 

As background, my problems first arose when I simply tried to click the 'send password reset' button from the firebase console, under the contextual menu for each user. This seems to be as far as I've made it, attempting to send the email programmatically. Any clues/hints/guesses offered would be greatly appreciated.

<img width="1440" alt="screen shot 2016-11-29 at 8 15 01 am" src="https://cloud.githubusercontent.com/assets/17304773/20715421/63741ebe-b60c-11e6-9d83-dd4f71c367c0.png">
<img width="1440" alt="screen shot 2016-11-29 at 8 15 08 am" src="https://cloud.githubusercontent.com/assets/17304773/20715423/63876816-b60c-11e6-9185-0961c90f593f.png">
<img width="1440" alt="screen shot 2016-11-29 at 8 15 15 am" src="https://cloud.githubusercontent.com/assets/17304773/20715422/63869b52-b60c-11e6-81fc-b92dd409dd42.png">

I apologize for the poorly formatted code, below, I'm not sure how to add returns and tabs.

`function sendPasswordReset() {
      var email = document.getElementById('email').value;
      // [START sendpasswordemail]
      firebase.auth().sendPasswordResetEmail(email).then(function() {
        // Password Reset Email Sent!
        // [START_EXCLUDE]
        alert('Password Reset Email Sent!');
        // [END_EXCLUDE]
      }).catch(function(error) {
        // Handle Errors here.
        var errorCode = error.code;
        var errorMessage = error.message;
        // [START_EXCLUDE]
        if (errorCode == 'auth/invalid-email') {
          alert(errorMessage);
        } else if (errorCode == 'auth/user-not-found') {
          alert(errorMessage);
        }
        console.log(error);
        // [END_EXCLUDE]
      });
      // [END sendpasswordemail];
    }`

**Comment by TMSCH** [comment]
Hi @itispeach, it doesn't seem that you are using FirebaseUI. For questions related to Firebase SDK, please use [StackOverflow](http://stackoverflow.com/questions/tagged/firebase-authentication) tagging `firebase-authentication`. More people will be able to help you there.

It seems that it may be a configuration related issue, so when you move your question to [StackOverflow](http://stackoverflow.com/questions/tagged/firebase-authentication), please also explain how you customize your email domain, it may help.

BTW, to do block code spanning several lines, you should put 3 backticks (```) before and after your code snippet ;). I think GitHub also provide a UI button to create them.

**Comment by itispeach** [comment]
@TMSCH Salut Thomas! (francophone ici). Thanks for your response. I didn't actually mean to use the SDK — I was hoping to use the Firebase UI. I just realized I was hiding the "having trouble" button in my implementation. Now that's showing again, here's what happens when I try to use it, and this is why I hid it from my users. As you can see in the screenshot below, I'm getting "Error code: auth/internal-error." Not sure, but since I added a custom domain for sending emails, do I need to update the Action URL? If so, to what? The hint in the firebase console reads "Action URL should link to an interface for users to complete tasks (sign up confirmation, password resets, email address changes)." As of right now, I simply dropped the code for firebaseUI into my web page. Will firebaseUI give me "an interface for users to complete tasks?" Any point in the right direction would be greatly appreciated.

<img width="1440" alt="screen shot 2016-11-30 at 7 38 45 am" src="https://cloud.githubusercontent.com/assets/17304773/20756610/8c3068fc-b6d0-11e6-959e-28b910978b23.png">

**Comment by TMSCH** [comment]
Salut! :) You shouldn't have to change the Action URL. The default points to a widget hosted by Firebase that includes all the handlers required. It is the recommended option.

Could you post the Network logs (there's a Network tab in the dev console), especially the one network request that fails? (I assume there should be one).

**Comment by itispeach** [comment]
Thank you for your time, Thomas. Here's a screenshot of what the network tab looks like :

<img width="1440" alt="screen shot 2016-12-02 at 11 15 06 pm" src="https://cloud.githubusercontent.com/assets/17304773/20857332/50532af4-b8f6-11e6-829d-5b8977313350.png">

I drilled down to more information. It only made sense to take a screenshot of the "Timing" section under  And here's a look, drilling down

Maybe there's an easier / prettier way to get this information to you, but I opted for copy & paste. You'll notice I left out the details under the "Header" sub-section of the Network tab. Because I'm generally clueless, I decided I would leave it out, until you ask for them/reassure me that it's okay to share those details in this venue. Below, you'll see the details of the Preview and Responses sub-tabs, pasted below : 

Thank you. — Matthew

[PREVIEW]
————————
{,…}
error
:
{errors: [{domain: "global", reason: "invalid", message: "INVALID_EMAIL : Missing sender"}], code: 400,…}
code
:
400
errors
:
[{domain: "global", reason: "invalid", message: "INVALID_EMAIL : Missing sender"}]
0
:
{domain: "global", reason: "invalid", message: "INVALID_EMAIL : Missing sender"}
domain
:
"global"
message
:
"INVALID_EMAIL : Missing sender"
reason
:
"invalid"
message
:
"INVALID_EMAIL : Missing sender"


[RESONSES]
————————
{
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "invalid",
    "message": "INVALID_EMAIL : Missing sender"
   }
  ],
  "code": 400,
  "message": "INVALID_EMAIL : Missing sender"
 }
}

**Comment by itispeach** [comment]
Also, I haven't brought this to your attention, but my website is a squarespace site. I'm dropping the code in approved spaces where they allow it, but I wonder if it's making things harder.

**Comment by itispeach** [comment]
As a small update : I decided to use javascript to create the 'firebaseui-auth-container' div and append it to the body of my page. This way all of my code was in the same spot in squarespace (and hopefully being embedded in the page together at the same time. Even in this scenario I am getting the same problem.

**Comment by bojeil-google** [comment]
Hey itispeach,
Before I escalate this issue, can you check your email templates in the Firebase Console and double check you didn't modify or remove the sender email from your email templates? You could have also provided an invalid email with an invalid domain the process which is causing this problem.

**Comment by itispeach** [comment]
Fair question! It is definitely there and it was also verified. As background, the mail client is Keiros (some implementation of a Microsoft Exchange server for the client I’m working with).

Matthew

**Comment by wuyanna** [comment]
Hi Matthew, can you try sending email in the Firebase console and see if it works? The cause of this error message is that there is some invalid value in the sender address/domain in the Firebase console.

**Comment by wuyanna** [comment]
It shouldn't have anything to do with Action URL. Can you check that the local part of the email sender (the part before @doamin) only contains alphabetic and numeric characters? Or change it to 'noreply' and see if that works? If that still doesn't work, let us know the project id and we can check for you

**Comment by itispeach** [comment]
Checked and tried. I'm still getting the same error. To confirm : I am now using {'noreply@{thedefaultfirebasedomain.com}. I'll send you the project ID directly.

**Comment by itispeach** [comment]
@wuyanna How do I message you directly with the Project ID — is this something I need to keep sensitive and out of this thread?

**Comment by wuyanna** [comment]
You can send your information to wuyanna@google.com, or add me as an editor to your project.

**Comment by itispeach** [comment]
Here is my App ID : brilliant-inferno-3808
You are also now an Editor.
Thank you very much.

Matthew

**Comment by wuyanna** [comment]
Hi Matthew, so the issue is that there is an invalid character comma (,) in the Sender name. Since the sender name is also a part of the sender address, it needs to follow the email address standard, and I think comma is a reserved token. After I removed the the comma in Sender name, it seems the issue doesn't appear anymore.

**Comment by itispeach** [comment]
Woohoo! You rock! This has been killing me!
It just worked for me as well. I can’t thank you enough for figuring that out.
I’ll re-update my DNS records and see if my custom domain doesn’t work as well, as long as I keep the sender name the same. Sincerely, thank you for taking the time to help me,

Matthew

**Comment by wuyanna** [comment]
My pleasure. Thanks for using Firebase :)

**Comment by Viral-Inc** [comment]
Exactly the same thing had been driving me crazy. Thank you for having this issue before me :)

**Comment by jpcarnes** [comment]
@wuyanna thanks for this thread, just ran into this issue and it was solved right away.

For anyone new to the thread make sure that you don't have a ',' in the sender field in the firebase template as seen below. That comma being present will cause the issue, in order to fix this I had to remove the comma before 'Inc.'

The same thing also happens if you try to use the firebase-email-trigger extension with a ',' in the sender.

![Screen Shot 2022-01-24 at 6 35 48 PM](https://user-images.githubusercontent.com/35235270/150888504-86b1784d-218d-4be3-8726-fd11c03ce4f1.png)
